Different schema support

Sunib opened the pull request:
I had the same problem is described in [this thread](https://github.com/mrodrig/json-2-csv/issues/12#issuecomment-110341395), it was a bit difficult in my situation to make all document schemas equal so I thought that it would be usefull to just add the different schema support. I had to make more changes then I though, but all tests (including the new one) are green now.

I added a seperate schema-combiner module that merges different schemas together.

The default is still that schema differences are not accepted. I added an option that people can enable to make this work.

Looking forward to your feedback on this!

Best regards,
Simon Koudijs
Intellifi

mrodrig commented:
Sorry, I haven't had a chance to look at this yet, but definitely intend to. I've been extremely busy and hope to get to this very soon.  Thanks!

Sunib commented:
Hi Mike,

I added extra unit tests and fixed a bug and my patch is now also able to handle equally named fields with different data types (see unit tests). 

No problem that you are busy, it's possible to pull this version with npm by adding this line into the package.json: `"json-2-csv": "https://github.com/sunib/json-2-csv/tarball/master"`

Best regards,
Simon

mrodrig commented:
Should be all set in 1.4.0 via #31. Thanks for implementing this.  I was able to simplify it a bit (~165 lines added instead of ~707), but I only added this because you brought the need to my attention.  Thanks again and sorry for the delay!
